lib/src/library_scanner.dart

```diff
@@ -110,25 +84,17 @@ class LibraryScanner {
       }
     }
 
-    UriResolver packageResolver = new PackageMapUriResolver(
-        PhysicalResourceProvider.INSTANCE, packageMap);
-
-    var resolvers = [
-      new DartUriResolver(sdk),
-      new ResourceUriResolver(PhysicalResourceProvider.INSTANCE),
-      packageResolver,
-    ];
-
-    AnalysisEngine.instance.processRequiredPlugins();
-
-    var options = new AnalysisOptionsImpl()..analyzeFunctionBodies = false;
+    var contextLocator = new ContextLocator();
+    var roots = contextLocator.locateRoots(includedPaths: [packagePath]);
+    if (roots.length != 1) {
+      throw new StateError('Expected exactly one context root, got $roots');
+    }
 
-    var context = AnalysisEngine.instance.createAnalysisContext()
-      ..analysisOptions = options
-      ..sourceFactory = new SourceFactory(resolvers);
+    var analysisContext = new ContextBuilder()
+        .createContext(contextRoot: roots.single, sdkPath: sdkPath);
 
     return new LibraryScanner._(
-        package, packagePath, packageResolver, context, overrides);
+        package, packagePath, analysisContext.currentSession, overrides);
```

Comment:
Does this mean we don't have any global-scoped cache in `package:analyzer`?

Reply:
Yes, `package:analyzer` does not cache any resolution or ASTs. Once analysis session is gone, all this data is gone too. There is a global file content cache, but it is keyed in `Expando` with the `FileContentOverlay` (and file system, but it is always the same). We create a new `FileContentOverlay` instance for each new `AnalysisContext`, so it goes away too.